corehq/apps/reports/models.py

```diff
@@ -1097,3 +1099,21 @@ def remove_custom_export(cls, domain, export_id):
     def clear_caches(self):
         super(HQGroupExportConfiguration, self).clear_caches()
         self.by_domain.clear(self.__class__, self.domain)
+
+
+class ReportsSidebarOrdering(models.Model):
+    domain = models.CharField(
+        max_length=256,
+        null=False,
+        blank=False,
+        unique=True
+    )
+    config = JSONField(
+        default=list,
+        help_text=(
+            "An array of arrays. Each array represents a heading in the sidebar navigation. "
+            "The first item in each array is a string, which will be the title of the heading. The second item in "
+            "the array is another array, each item of which is the name of a report class. Each of these reports "
+            "will be listed under the given heading in the sidebar nav."
+        )
+    )
```

Comment:
Decided to just go with a "manual" check of the structure here: https://github.com/dimagi/commcare-hq/pull/14963/commits/9c1e981cc82e5432f5266831cadb2f180abd8a4e.
Since my config structure is nested lists instead of nested objects, I think jsonobject won't actually be that useful for validating. For example I don't think there is a way in jsonobject to specify that a top level JsonObject should be a list, or that the first item of a list should have certain properties while the second should some other properties.

I suppose I could change the config structure to be object based instead of list based, but given that this isn't going to be a widely used feature, I'm inclined to think it's not worth restructuring. (Plus it would be more tedious to write the config).